pkg/io/video/scaleycrcb.go

```diff
@@ -0,0 +1,39 @@
+package video
+
+import (
+	"image"
+	"image/color"
+)
+
+type rgbLikeYCbCr struct {
+	y  *image.Gray
+	cb *image.Gray
+	cr *image.Gray
+}
+
+func (p *rgbLikeYCbCr) ColorModel() color.Model {
+	return color.RGBAModel
+}
+
+func (p *rgbLikeYCbCr) Bounds() image.Rectangle {
+	return p.y.Rect
+}
+
+func (p *rgbLikeYCbCr) At(x, y int) color.Color {
+	var yy, cb, cr uint8
+	yy = p.y.GrayAt(x, y).Y
+	if (image.Point{x, y}.In(p.cb.Rect)) {
```

Comment:
I think we'll lose some important pixels. It's better to scale down the x and y accordingly for chroma planes. 

Something like,

```
// probably store these in struct instead
x = x * p.y.Dx() / p.cb.Dx()
y = y * p.y.Dy() / p.cb.Dy() 

cb = p.cb.GrayAt(x, y).Y
cr = p.cr.GrayAt(x, y).Y
```
What do you think?

Reply:
It is used to process three gray images (which are the planes of a YCbCr image) at once by one scaling factor, without having overhead of `Gray->RGBA->Gray` color transform in `image/draw`'s Scaler.
(`image.YCbCr` don't have `Set()`, so it can't be directly passed to `Scaler.Scale()`.)
Pixel buffer of `p.cb` and `p.cr` are backended directly by original `YCbCr`'s `Cb`, `Cr` buffer (at [here](https://github.com/pion/mediadevices/pull/55/files#diff-004d7e97caf7d4018cb1de5c46028d98R157-R163)) which actually have smaller size, so all raw pixels in Cb/Cr layers are used.

Reply:
Right, since Cb/Cr layers are smaller, the if statement will be false if we try to go beyond Cb/Cr sizes but still lower than Y size. 

Let say we have a 16x16 image and a 4:2:2 subsampling, we should be able to say `At(15, 15)` and get the Cb/Cr values at 15,15. But, since the Cb/Cr rectangles are just 16x8, the if statement will be false.

Reply:
This image is not handled as YCbCr, but independent 3 Gray images.
For example of 4x4-pixels 4:2:2 image scaled by 1/2 by nearest neighbor,
input image:
```
Y 4x4
1 1 0 0
1 1 0 0
0 0 2 2
0 0 2 2
Cb 4x2
4 4 2 2
4 4 2 2
Cb 4x2
6 6 3 3
6 6 3 3
```
intermediate rgbLikeYCbCr:
```
R 4x4
1 1 0 0
1 1 0 0
0 0 2 2
0 0 2 2
G 4x4
4 4 2 2
4 4 2 2
P P P P // virtually padded, actually out of the capacity
P P P P
B 4x4
6 6 3 3
6 6 3 3
P P P P
P P P P
```
Scaled rgbLikeYCbCr:
```
R 2x2
1 0
0 2
G 2x2
4 2
P P
B 2x2
6 3
P P
```
Reconstructed YCbCr:
```
Y 2x2
1 0
0 2
Cb 2x1
4 2
Cr 2x1
6 3
```

Reply:
Ah, I got you. The paddings are not actually there. But, we virtually created them by checking if they're in the boundaries. If not, then we'll just return 0 (padding value).